File: src/main/kotlin/com/manager/labo/view/MainPanel.java
package com.manager.labo.view;

import com.manager.labo.utils.ActionCommand;
import com.manager.labo.view.components.JPanelEnchancer;
import javax.swing.JButton;
import javax.swing.JPanel;

public class MainPanel extends JPanel {
    @ActionCommand("Examination-Add")
    private JButton newRequest;
    @ActionCommand("Patient-List")
    private JButton patientsLists;
    @ActionCommand("Examination-List")
    private JButton examinationList;

    private final void createUIComponents() {
        newRequest = new JButton();
        patientsLists = new JButton();
        examinationList = new JButton();
        (new JPanelEnchancer(this)).standardActions().initButtonsActionCommands();
    }
}

File: src/main/kotlin/com/manager/labo/view/MainPanel.kt
package com.manager.labo.view

import com.manager.labo.utils.ActionCommand
import com.manager.labo.utils.EXAMINATION_ADD
import com.manager.labo.utils.EXAMINATION_LIST
import com.manager.labo.utils.PATIENT_LIST
import com.manager.labo.view.components.JPanelEnchancer
import javax.swing.JButton
import javax.swing.JPanel

class MainPanel : JPanel() {
    @ActionCommand(EXAMINATION_ADD)
    private var newRequest: JButton? = null

    @ActionCommand(PATIENT_LIST)
    private var patientsLists: JButton? = null

    @ActionCommand(EXAMINATION_LIST)
    private var examinationList: JButton? = null
    private fun createUIComponents() {
        newRequest = JButton()
        patientsLists = JButton()
        examinationList = JButton()
        JPanelEnchancer(this).standardActions().initButtonsActionCommands()
    }
}
